Core_Pregnancy/PregnancyGameController.cs

```diff
@@ -0,0 +1,303 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ExtensibleSaveFormat;
+#if KK
+    using ActionGame;    
+#endif
+using KKAPI.MainGame;
+using Manager;
+using UnityEngine;
+using Random = UnityEngine.Random;
+#if AI
+    using AIChara;
+    using AIProject;
+    using AIProject.SaveData;
+#endif
+
+namespace KK_Pregnancy
+{
+    public class PregnancyGameController : GameCustomFunctionController
+    {
+        #if KK
+            private static readonly HashSet<SaveData.Heroine> _startedPregnancies = new HashSet<SaveData.Heroine>();
+        #elif AI
+            private static readonly HashSet<AgentData> _startedPregnancies = new HashSet<AgentData>();
+        #endif
+
+        internal static bool InsideHScene { get; private set; }
+
+
+        #if KK
+        
+            protected override void OnDayChange(Cycle.Week day)
+            {
+                // Use Sunday for weekly stuff because it is always triggered (all other days can get skipped)
+                if (day == Cycle.Week.Holiday)
+                {
+                    // At start of each week increase pregnancy week counters of all pregnant characters
+                    ApplyToAllDatas((heroine, data) => AddPregnancyWeek(data));
+                }                                
+            }
+
+        #elif AI
+
+            protected override void OnDayChange(int day)
+            {
+                // 1 day = 1 week in AI
+                // At start of each week increase pregnancy week counters of all pregnant characters
+                ApplyToAllDatas((heroine, data) => AddPregnancyWeek(data));
+            }
```

Comment:
Im making 1 day in AI == 1 week of pregnancy, since you can't skip weeks like in KK